File: src/FromList.java
package org.nlogo.extensions.string;

import org.nlogo.api.Argument;
import org.nlogo.api.Context;
import org.nlogo.api.DefaultReporter;
import org.nlogo.api.ExtensionException;
import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;
import org.nlogo.api.LogoListBuilder;
import org.nlogo.api.LogoList;

public class FromList extends DefaultReporter{

    @Override
    public Syntax getSyntax(){
        return Syntax.reporterSyntax(new int[] { Syntax.ListType() , Syntax.StringType()},
                                     Syntax.StringType());
    }

    @Override
    public Object report(Argument[] arg0, Context arg1)
             throws ExtensionException, LogoException {
        LogoList input = arg0[0].getList();
        String delimiter = arg0[1].getString();
        StringBuilder sb = new StringBuilder();
        for (Object o : input){
            sb.append(org.nlogo.api.Dump.logoObject(o));
            sb.append(delimiter);
        }
        return sb.toString();
    }

}

File: src/FromList.scala
package org.nlogo.extensions.string

import org.nlogo.api.Argument
import org.nlogo.api.Context
import org.nlogo.api.Reporter
import org.nlogo.api.ExtensionException
import org.nlogo.api.LogoException
import org.nlogo.core.Syntax
import org.nlogo.api.LogoListBuilder
import org.nlogo.core.LogoList

class FromList extends Reporter {

    override def getSyntax(): Syntax = {
      Syntax.reporterSyntax(
        right = List (Syntax.ListType),
        ret   = Syntax.StringType)
    }

   @throws(classOf[ExtensionException])
   @throws(classOf[LogoException])
    override def report(args: Array[Argument], context: Context): Object = {
      val input: LogoList = args(0).getList
      val sb: StringBuilder = new StringBuilder
      input.foreach(sb ++= org.nlogo.api.Dump.logoObject(_))
      sb.toString
    }

}
